[DependencyInjection] Cast tag attribute value to string

ruudk opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  |no 
| Deprecations? |no 
| Tickets       | 
| License       | MIT
| Doc PR        | 

```
DOMElement::setAttribute(): Passing null to parameter #2 ($value) of type string is deprecated
```
This happens when a tag value is `null` on PHP 8.1.

derrabus commented on src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php:
```diff
@@ -138,7 +138,7 @@ private function addService(Definition $definition, ?string $id, \DOMElement $pa
                 $tag->setAttribute('name', $name);
                 foreach ($attributes as $key => $value) {
-                    $tag->setAttribute($key, $value);
+                    $tag->setAttribute($key, (string) $value);
```
Let's keep this change as small as possible.

```suggestion
                    $tag->setAttribute($key, $value ?? '');
```

ruudk replied:
Fine by me, I choose for casting because that is what's done in the rest of the file.

ruudk committed:
[DependencyInjection] Cast tag value to string

DOMElement::setAttribute(): Passing null to parameter #2 ($value) of type string is deprecated

This happens when a tag value is `null` on PHP 8.1.

fabpot commented:
Thank you @ruudk.
